tests/components/tankerkoenig/test_config_flow.py

```diff
@@ -222,6 +222,59 @@ async def test_import(hass: HomeAssistant):
     assert mock_setup_entry.called
 
 
+async def test_reauth(hass: HomeAssistant):
+    """Test starting a flow by user to re-auth."""
+
+    mock_config = MockConfigEntry(
+        domain=DOMAIN,
+        data={**MOCK_USER_DATA, **MOCK_STATIONS_DATA},
+        unique_id=f"{MOCK_USER_DATA[CONF_LOCATION][CONF_LATITUDE]}_{MOCK_USER_DATA[CONF_LOCATION][CONF_LONGITUDE]}",
+    )
+    mock_config.add_to_hass(hass)
+
+    with patch(
+        "homeassistant.components.tankerkoenig.async_setup_entry", return_value=True
+    ) as mock_setup_entry, patch(
+        "homeassistant.components.tankerkoenig.config_flow.getNearbyStations",
+    ) as mock_nearby_stations:
+        # re-auth initialized
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_REAUTH, "entry_id": mock_config.entry_id},
+            data=mock_config.data,
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "reauth_confirm"
+
+        # re-auth unsuccessful
+        mock_nearby_stations.return_value = {"ok": False}
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input={
+                CONF_API_KEY: "269534f6-xxxx-xxxx-xxxx-yyyyzzzzxxxx",
+            },
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "reauth_confirm"
+        assert result["errors"] == {CONF_API_KEY: "invalid_auth"}
+
+        # re-auth successful
+        mock_nearby_stations.return_value = MOCK_NEARVY_STATIONS_OK
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input={
+                CONF_API_KEY: "269534f6-xxxx-xxxx-xxxx-yyyyzzzzxxxx",
```

Comment:
We should use a different key than the original key so that we can test that it's updated.

Reply:
for sure, you are right 🙈